### PR TITLE
Task 15.2 Avsiievych endpoint update workspace

### PR DESCRIPTION
**Pull request by papich3pich**

### CODE REVIEW

File: Provis.Core/Interfaces/Services/IWorkspaceService.cs [hunk 1]
```diff
@@ -16,5 +16,7 @@ public interface IWorkspaceService
         Task<List<WorkspaceInfoDTO>> GetWorkspaceListAsync(string userid);
 
         Task SendInviteAsync(InviteUserDTO inviteUser, string userId);
+
+        Task UpdateWorkspaceAsync(WorkspaceUpdateDTO workspaceDTO, string userid);
```
Comment (YevheniiVovk) [inline Provis.Core/Interfaces/Services/IWorkspaceService.cs]: naming userId

File: Provis.WebApi/Policy/WorkspaceRolesAuthorizationHandler.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ protected override Task HandleRequirementAsync(AuthorizationHandlerContext conte
                     bodyStr = reader.ReadToEnd();
 
                     JObject jObj = JObject.Parse(bodyStr);
-                    workspaceId = (int)jObj["workspaceId"];
+                    workspaceId = (int?)jObj["workspaceId"];
```
Comment (YevheniiVovk) [inline Provis.WebApi/Policy/WorkspaceRolesAuthorizationHandler.cs]: Can we get null pointer exception here ?

Comment (papich3pich) [reply Provis.WebApi/Policy/WorkspaceRolesAuthorizationHandler.cs]: Actually, with proper body in POST/PUT request, we can't get it

File: Provis.Core/Validation/WorkspaceUpdateValidation.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public class WorkspaceUpdateValidation : AbstractValidator<WorkspaceUpdateDTO>
+    {
+        public WorkspaceUpdateValidation()
+        {
+            RuleFor(workspace => workspace.Name)
+                .NotEmpty()
+                .NotNull()
+                .Length(1, 50)
+                .Matches("^[^£# “”]*$").WithMessage("{PropertyName} must not contain the following characters £ # “” or spaces.");
```
Comment (YevheniiVovk) [inline Provis.Core/Validation/WorkspaceUpdateValidation.cs]: Let's move .WithMessage to new line to be consistency with previous

File: Provis.Core/Validation/WorkspaceCreateValidation.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public class WorkspaceCreateValidation : AbstractValidator<WorkspaceCreateDTO>
+    {
+        public WorkspaceCreateValidation()
+        {
+            RuleFor(workspace => workspace.Name)
+                .NotEmpty()
+                .NotNull()
+                .Length(1, 50)
+                .Matches("^[^£# “”]*$").WithMessage("{PropertyName} must not contain the following characters £ # “” or spaces.");
```
Comment (YevheniiVovk) [inline Provis.Core/Validation/WorkspaceCreateValidation.cs]: Same here

File: Provis.Core/Services/WorkspaceService.cs [hunk 5]
```diff
@@ -68,6 +68,24 @@ public async Task CreateWorkspaceAsync(WorkspaceCreateDTO workspaceDTO, string u
...
         }
+        public async Task UpdateWorkspaceAsync(WorkspaceUpdateDTO workspaceUpdateDTO, string userId)
+        {
+            var user = await _userManager.FindByIdAsync(userId);
+
+            _ = user ?? throw new HttpException(System.Net.HttpStatusCode.NotFound, "User with Id not exist");
+            
+            var workspaceRec = await _workspaceRepository.GetByKeyAsync(workspaceUpdateDTO.WorkspaceId);
+
+            _ = workspaceRec ?? throw new HttpException(System.Net.HttpStatusCode.NotFound, "Workspace with with Id not found");
```
Comment (YevheniiVovk) [inline Provis.Core/Services/WorkspaceService.cs]: Length should be low then 100 characters

File: Provis.Core/Validation/WorkspaceUpdateValidation.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+            RuleFor(workspace => workspace.Description)
+                .NotNull()
+                .Length(0, 100)
+                .Matches("^[^£# “”]*$").WithMessage("{PropertyName} must not contain the following characters £ # “” or spaces.");
```
Comment (YevheniiVovk) [inline Provis.Core/Validation/WorkspaceUpdateValidation.cs]: Same here

File: Provis.Core/Services/WorkspaceService.cs (hunk 5 above)
Comment (papich3pich) [reply Provis.Core/Services/WorkspaceService.cs]: Is this comment about exactly this part of code or maybe about similar to update validation, create validation code?

Comment (YevheniiVovk) [reply Provis.Core/Services/WorkspaceService.cs]: 79 line contains more then 100 characters please split it to two lines by , as separator

File: Provis.Core/Validation/WorkspaceCreateValidation.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    public class WorkspaceCreateValidation : AbstractValidator<WorkspaceCreateDTO>
+    {
+        public WorkspaceCreateValidation()
+        {
+            RuleFor(workspace => workspace.Name)
+                .NotEmpty()
+                .NotNull()
+                .Length(1, 50)
+                .Matches("^[^£# “”]*$").
+                WithMessage("{PropertyName} must not contain the following characters £ # “” or spaces.");
```
Comment (YevheniiVovk) [inline Provis.Core/Validation/WorkspaceCreateValidation.cs]: .WithMessage should be from new line the character dot should be from new line too

File: Provis.Core/Validation/WorkspaceCreateValidation.cs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+            RuleFor(workspace => workspace.Description)
+                .NotNull()
+                .Length(0, 50)
+                .Matches("^[^£# “”]*$").
+                WithMessage("{PropertyName} must not contain the following characters £ # “” or spaces.");
```
Comment (YevheniiVovk) [inline Provis.Core/Validation/WorkspaceCreateValidation.cs]: Same as privios

**Review by YevheniiVovk — APPROVED** [review]

**Review by AndriyKovalets — APPROVED** [review]

**Pull request merged**